internal/controllers/snapshot_controller.go

```diff
@@ -169,8 +169,22 @@ func (r *SnapshotReconciler) removeSnapshot(log logr.Logger, snapshotID string,
 	}
 	log.V(2).Info("Snapshot references", "pools", len(pools), "rbd-images", len(imgs))
 
-	if len(pools) != 0 || len(imgs) != 0 {
-		return fmt.Errorf("unable to delete snapshot: still in use")
+	for _, clonedImgName := range imgs {
+		log.V(2).Info("Open cloned image to flatten", "ClonedImageID", clonedImgName)
+		img, err := librbd.OpenImage(ioCtx, clonedImgName, librbd.NoSnapshot)
```

Comment:
`img` in the loop here shadows the above function parameter `img` variable. Please use  different variable name in the loop.

```suggestion
		clonedImg, err := librbd.OpenImage(ioCtx, clonedImgName, librbd.NoSnapshot) 
```